content/en/docs/plugin-guide/plugin-policy-engine/packages/spinnaker.deployment/tasks.before/cleanupArtifacts.md

```diff
@@ -0,0 +1,358 @@
+                    ],
+                    "image": "nginx/nginx-prometheus-exporter:0.3.0",
+                    "imagePullPolicy": "Always",
+                    "name": "nginx-exporter",
+                    "ports": [
+                      {
+                        "containerPort": 9113,
+                        "name": "nginx-ex-port",
+                        "protocol": "TCP"
+                      }
+                    ],
+                    "resources": {},
+                    "terminationMessagePath": "/dev/termination-log",
+                    "terminationMessagePolicy": "File"
+                  }
+                ],
+                "dnsPolicy": "ClusterFirst",
+                "restartPolicy": "Always",
+                "schedulerName": "default-scheduler",
+                "securityContext": {},
+                "terminationGracePeriodSeconds": 30,
+                "volumes": [
+                  {
+                    "configMap": {
+                      "defaultMode": 420,
+                      "name": "nginx-status-conf-v000"
+                    },
+                    "name": "nginx-status-conf"
+                  }
+                ]
+              }
+            }
+          },
+          "status": {
+            "availableReplicas": 10,
+            "conditions": [
+              {
+                "lastTransitionTime": "2021-04-30T21:19:15Z",
+                "lastUpdateTime": "2021-05-07T20:39:03Z",
+                "message": "ReplicaSet \"hostname-f5b957cc\" has successfully progressed.",
+                "reason": "NewReplicaSetAvailable",
+                "status": "True",
+                "type": "Progressing"
+              },
+              {
+                "lastTransitionTime": "2021-05-07T22:39:40Z",
+                "lastUpdateTime": "2021-05-07T22:39:40Z",
+                "message": "Deployment has minimum availability.",
+                "reason": "MinimumReplicasAvailable",
+                "status": "True",
+                "type": "Available"
+              }
+            ],
+            "observedGeneration": 10,
+            "readyReplicas": 10,
+            "replicas": 10,
+            "updatedReplicas": 10
+          }
+        }
+      ]
+    }
+  }
+}
+```
+</details>
+
+## Example Policy
+Prevents cleanupArtifacts tasks from running on any account in a predefined list.
+{{< prism lang="rego" line-numbers="true" >}}
+package spinnaker.deployment.tasks.before.cleanupArtifacts
+
+productionAccounts :=["prod1","prod2"]
+
+deny["Artifactss may not be cleaned up from production accounts"] { 
+	input.deploy.account==productionAccounts[_]
+}
+
+{{< /prism >}}
+
+## Keys
+
+| Key                          | Type     | Description                                    |
+| ---------------------------- | -------- | ---------------------------------------------- |
+| `input.deploy.account`       | `string` | The spinnaker account being deployed to.       |
```

Comment:
CC @stephenatwell

Reply:
Spinnaker has a definition for 'accounts', each 'account' is associated with a cloud provider. On my personal spinnaker instance, for example, I have 3 'accounts', for a single 'kubernetes cluster', the different accounts have different namespaces and policies associated with them. In my kustomize repo I can see these accounts defined in 'patch-kube.yml' under the path spec.spinnakerConfig.config.providers.kubernetes.accounts[].name

Both the tamls, and the UI just call them 'accounts'. I'm fine calling thenm 'spinnaker accounts' or (in the case of stages that interact with kubernetes) 'kubernetes accounts'). Personally I think spinnaker accounts, or just 'accounts' is safer because we have stages that talk to multiple cloud providers, and specify the cloud provider by choosing the account.

Reply:
@brian-armory thoughts?

Reply:
I'd go with `accounts`